### PR TITLE
Bug/dl 62

### PR DESCRIPTION
**Pull request by Jahenstein**
Fixed email test, modified files:
* .env - added fake email credenctials
* tests/conftest.py - added TESTING = True, so emails are not send 
* forms/registration_form.py - deleted, unnecessary form (duplicate)
* **models/decorators_roles.py** and **tests/test_decorators_roles.py** - files from different task

### CODE REVIEW

File: config.py [hunk 1]
```diff
@@ -15,12 +15,14 @@ class Config(object):
...
     ADMINS = [getenv('MAIL_USERNAME')]
 
+
 class DevConfig(Config):
     SECRET_KEY = '4f\g45t45gfjerkfefker'
     SECURITY_PASSWORD_SALT = 'my_precious_two'
     SQLALCHEMY_TRACK_MODIFICATIONS = False
     SQLALCHEMY_POOL_TIMEOUT = 30
     DEBUG = True
+    TESTING = True
```
Comment (etude-ist) [inline config.py]: Why you added that?

File: models/decorators_roles.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+from functools import wraps
+from models.users import User
+
+
+def require_logged_in(redirect_page='library.login'):
+    def decorator(func):
+        @wraps(func)
+        def inner_func(*args, **kwargs):
+            if 'logged_in' in session:
+                if session['logged_in']:
```
Comment (etude-ist) [inline models/decorators_roles.py]: I guess the two `if`s could be reduced to one `if`.

File: tests/test_decorators_roles.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
+from models.decorators_roles import (
+    require_logged_in,
+    require_not_logged_in,
+    require_role
+)
+import pytest
+from models.users import Role, RoleEnum
+from mimesis import Generic
+from flask import session
```
Comment (etude-ist) [inline tests/test_decorators_roles.py]: It's a minor style issue, but we prefer ordered, grouped imports in that project: python std lib, python libs, flask imports, flask libs and app modules imports.

File: tests/test_decorators_roles.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    @require_logged_in(redirect_page)
+    def func_logged_in():
+        return 'ok'
+
+    @require_not_logged_in(redirect_page)
+    def func_not_logged_in():
+        return 'ok'
+
+    if login:
```
Comment (etude-ist) [inline tests/test_decorators_roles.py]: Let's rewrite that using only asserts.

File: tests/test_decorators_roles.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+        role_admin = Role.query.filter_by(name=RoleEnum.ADMIN).first()
+        db_user.roles.append(role_admin)
+
+    session['id'] = db_user.id
+
+    @require_role(role=required_role)
+    def func_role():
+        return 'ok'
+
+    if role == required_role:
```
Comment (etude-ist) [inline tests/test_decorators_roles.py]: Let's rewrite that using only asserts.

File: tests/test_decorators_roles.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        role_admin = Role.query.filter_by(name=RoleEnum.ADMIN).first()
+        db_user.roles.append(role_admin)
+
+    session['id'] = db_user.id
+
+    @require_role(role=required_role, redirect_page=redirect_page)
+    def func_role():
+        return 'ok'
+
+    if role == required_role:
```
Comment (etude-ist) [inline tests/test_decorators_roles.py]: As previous comment.

File: config.py (hunk 1 above)
Comment (Jahenstein) [reply config.py]: With that condition app don't send mail but only mimics it. That's why it's possible do give false credentials in .env without causing error.

File: models/decorators_roles.py (hunk 2 above)
Comment (Jahenstein) [reply models/decorators_roles.py]: Roger that

File: config.py (hunk 1 above)
Comment (etude-ist) [reply config.py]: We should have separate config for testing and don't mix that with dev config.

**Pull request merged**